File: ReverseArray-InPlace/ArrayReverse.java
import java.util.*;

class ArrayReverse
{
    public static int __(int x, int y) {return x;}

    /* Function to reverse arr[]
    from start to end*/
    public static void revereseArray(int []arr,
                                     int n)
    {
        for (int i = 0; i < n / 2; i++)
            arr[i] = __(arr[n - i - 1],
                        arr[n - i - 1] = arr[i]);
    }
    public static void main(String[] args)
    {
        int []arr = new int[]{1, 2, 3, 4, 5, 6};
        revereseArray(arr, arr.length);
        System.out.println("Reversed array is");
        for (int i = 0; i < arr.length; i++)
            System.out.print(Integer.toString(arr[i]) + " ");
    }
}

File: ReverseArray-InPlace/ArrayReverse.kt
import java.util.*

internal object ArrayReverse {
    fun __(x: Int, y: Int): Int {
        return x
    }

    /* Function to reverse arr[]
    from start to end*/
    fun revereseArray(arr: IntArray,
                      n: Int) {
        for (i in 0 until n / 2)
            arr[i] = __(arr[n - i - 1],
                    arr[n - i - 1] = arr[i])
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val arr = intArrayOf(1, 2, 3, 4, 5, 6)
        revereseArray(arr, arr.size)
        println("Reversed array is")
        for (i in arr.indices)
            print(Integer.toString(arr[i]) + " ")
    }
}
